odbc/src/api/data.rs

```diff
@@ -1152,7 +1144,11 @@ pub mod i16_len {
         let message = cstr::to_widechar_vec(message);
         let (len, ret) = set_output_wstring_helper(&message, output_ptr, buffer_len, None);
         // Only copy the length if the pointer is not null
-        ptr_safe_write(text_length_ptr, len.try_into().unwrap_or(SmallInt::MAX));
+        ptr_safe_write(
+            text_length_ptr,
+            len.try_into()
+                .expect("The data is too large to fit in the buffer on this platform."),
```

Comment:
```suggestion
                .expect("Data too large to fit"),
```
or `The data written is too large to describe the length on this platform.` like you changed in the other call above. Up to you. Try to not use `buffer` because it gets confusing with the buffer we want to write in itself.